### PR TITLE
Remove bootstrap.proj

### PR DESCRIPTION
**Pull request by jeffkl**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2186

Regression? Last working version:

## Description

This removes `bootstrap.proj` and replaces it with more retail MSBuild logic and allows a user to clone our repository and build it as long as they have Visual Studio and .NET SDK installed.

I had to re-recreate https://github.com/NuGet/NuGet.Client/pull/5070 since GitHub won't let me re-open it.  I will address the comments from it in this PR.

The main goal is to make it so nothing needs to be initialized, the act of building the repo is what pulls in all necessary dependencies.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: build/BuildValidator.proj
Comment (jeffkl) [inline build/BuildValidator.proj:1]: This replaces [BuildValidator.ps1](https://github.com/NuGet/NuGet.Client/blob/84d99ce064a5f70628c89af9f30693581e84dc37/scripts/cibuild/BuildValidator.ps1) which depended on bootstrap.proj to pull down a package (NuGetValidator) and on the VSIX already being built.  Now that its an MSBuild project, we will get binary logs from hosted builds and in this case I'm embedding the text log from NuGetValidator in the binary log so we'll get more logs than before.

File: build/common.project.props
Comment (jeffkl) [inline build/common.project.props:3]: MicroBuild is now imported in every project via CPM's `<GlobalPackageReference />`

https://github.com/NuGet/NuGet.Client/pull/5142/files#diff-5baf5f9e448ad54ab25a091adee0da05d4d228481c9200518fcb1b53a65d4156R129

File: src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj
Comment (jeffkl) [inline src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj:118]: The `ILMerge` package provides this MSBuild property for us automatically!

File: test/TestUtilities/CreateEndToEndTestPackage/CreateEndToEndTestPackage.proj
Comment (jeffkl) [inline test/TestUtilities/CreateEndToEndTestPackage/CreateEndToEndTestPackage.proj:1]: This replaces [CreateEndToEndTestPackage.ps1](https://github.com/NuGet/NuGet.Client/blob/84d99ce064a5f70628c89af9f30693581e84dc37/scripts/cibuild/CreateEndToEndTestPackage.ps1) as an MSBuild project so we can get binary logs from hosted builds.  It's also using my `Microsoft.Build.Artifacts` package which is able to stage artifacts very quickly since it has robocopy-like logic.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:417]: I'm assuming you validated that this still works correctly, as the path looks like a path where packages are extracted, so I find it surprising that there's no package version in the path.

**Review by zivkan — APPROVED** [review]

**Comment by zivkan** [comment]
Oh, please don't merge this until after M2 mode, to minimize risk of blocking urgent insertions

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (jeffkl) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:417]: Yes the assembly is now copied from `$(PkgMicrosoft_Web_Xdt)` to a folder that does not include the version so that none of our build logic needs to worry about the version.

**Review by zivkan — APPROVED** [review]

**Pull request merged**
